### ISSUE TITLE
Crash on unplugging display

### ISSUE THREAD

**Issue by jpalus** [Alexays/Waybar#1104]
waybar terminates on unplugging external screen. I've managed to track it down to:
https://github.com/Alexays/Waybar/blob/cf3d6545c36830eabba6f87766ab758b609cd266/src/client.cpp#L197
but am not proficient enough with C++ to tell what's the issue exactly. I suppose something's wrong with Bar disposal.

waybar: 0.9.7
arch: aarch64
gcc: 11.1.0

**Comment by misuto** [comment]
I have the same problem but it's triggered for me when i close the laptop lid which is configured in clamshell mode for use with a dock.

dist: Fedora 34
waybar: v0.9.7
arch: amd64

**Comment by andrewnicolalde** [comment]
I am also having this problem on Fedora 34, waybar version v0.9.7, Sway 1.6, on amd64.

**Comment by jpalus** [comment]
It appears to be caused by following code:
https://github.com/Alexays/Waybar/blob/7c1303f57cbf2e80005ac7650b2a38af8b2dc37e/include/util/sleeper_thread.hpp#L80-L81
Or at least with this line commented out I was not able to reproduce the issue (so far).

**Comment by albocc** [comment]
I am having the same issue and it seems like I can reproduce it every time. All I need to do is turn off my secondary display and I can watch waybar crash.

Here is the stack trace if it's of any use to you:
<details>
<summary>Stack trace</summary>

```
Process 86029 (waybar) of user 1000 dumped core.
Found module linux-vdso.so.1 with build-id: 9712ed076859a4223880db174a3db6aede4eb1a9
Found module libicudata.so.69 with build-id: 0ab994a49ef1848499c4af333b3266f28432a922
Found module libicuuc.so.69 with build-id: 5cf18c56e2f64efdac32cf61fb9c0c48e9bb1797
Found module libxml2.so.2 with build-id: fe447813db6ae9c70345a25ec4c7d79d82cabb0c
Found module librsvg-2.so.2 with build-id: 25ea7d4ce2ab875330c04c0e1c5b6830f81e565b
Found module libpixbufloader-svg.so with build-id: 532fccc97d22b5c93890c17efc1c6e44d09e066a
Found module libdconfsettings.so with build-id: 7a4ba7af122c4d11168004a9ea6def330f02089f
Found module libgpg-error.so.0 with build-id: ba85170c2d9343ea05eea8fa2048c212ff4ef552
Found module libgcrypt.so.20 with build-id: 4a114f6a371dee16ad6e01023b0b3c654b8582a3
Found module libcap.so.2 with build-id: d3d0ddf2f54f554fb280d7eb41a06dd39173009e
Found module liblz4.so.1 with build-id: e63600ab23b2f6997f42fac2fa56e1f02ce159a1
Found module libogg.so.0 with build-id: ca0cae30a809f10db1f16e54a23b4e5ff7b2c54c
Found module libvorbis.so.0 with build-id: ad0a6c1af8f219b4e314090b6e8847223d424cbe
Found module libopus.so.0 with build-id: 649d3ddab8764d7caf73648004a35cb13e23a5e1
Found module libFLAC.so.8 with build-id: e568dc3d434723c72d6d302eb0b2559f27e91faa
Found module libvorbisenc.so.2 with build-id: 28ed815b8b1d322e6bf2bfdce4b0af2789b74335
Found module libXdmcp.so.6 with build-id: 8ca0792d23c8b8b4c0864297512349292bea5955
Found module libXau.so.6 with build-id: 1c67764663e07bec24d8951e5fd93f4d165979ff
Found module libdatrie.so.1 with build-id: 6fe3b6ece2c8e7d11869fa051375128d8f808f58
Found module liblzma.so.5 with build-id: f8d607153b2b315f1aaa1c8e12553d544e661f2d
Found module libzstd.so.1 with build-id: 4b10444c1560ebc574af4d5f488b7408b22d450e
Found module libasyncns.so.0 with build-id: 6acff139d56b5139ce5eb50a66c62b215e1d332d
Found module libsystemd.so.0 with build-id: 15e863a1804ee74d4dd4c79ada848c6f1facb2d2
Found module libsndfile.so.1 with build-id: 72d5531e74ed3e9afe49e95e449c40cb8984a258
Found module libblkid.so.1 with build-id: ac40031740107dc1202fac27f75babfca7b5a8b9
Found module libatspi.so.0 with build-id: 046ee7a264956ff92020b0773ab583b23b93933c
Found module libbz2.so.1.0 with build-id: 919597c477c9b2cb9cdbb7745ed6494ac0e6da60
Found module libexpat.so.1 with build-id: 8850138eae6d9d4d43c5c4b2ac48393bc4279037
Found module libpixman-1.so.0 with build-id: 341f793dcada3a48a306a793d265a517e3f2e7d6
Found module libxcb-shm.so.0 with build-id: fb797f299a446f559a95afcc168227482cc800d1
Found module libxcb-render.so.0 with build-id: a37bdb37744b508be9dd29fd9ccb9bb0170d43e5
Found module libxcb.so.1 with build-id: 0d1ef11740a5daad2ee331e812a51aa6574af222
Found module libXrender.so.1 with build-id: 97e0b9ab6ba96ebc86527cc2b3c3078aad8616b3
Found module libgraphite2.so.3 with build-id: 47761dc11e553f519cde97ed9ee985be12ccdae2
Found module libthai.so.0 with build-id: 3fdce1a7db2b94d9168eea540dbf5ff5caa3b8ad
Found module libtiff.so.5 with build-id: e8d89a0fb0847641ddb75c5f03fa89d68155a75d
Found module libjpeg.so.8 with build-id: 2b2ee5daa1e21c0586419b05282528a6556259b2
Found module libpng16.so.16 with build-id: 2dc0bce07f199bf983c07a05fb95a6f4af83a9b3
Found module libbsd.so.0 with build-id: f872eb7bd22a103b7208ca9e4730ff8dc6d83590
Found module libasound.so.2 with build-id: 9288c92ff737b1999761df2018fcd0e7cbf46d5a
Found module libdl.so.2 with build-id: 5abc547e7b0949f89f3c0e21ab0c8331a7440a8a
Found module libdbus-1.so.3 with build-id: 74f2ab9c60512f3a93c932c3f627564d42e0b11e
Found module libpulsecommon-14.2.so with build-id: 1f49d299b8052285075d6443bd910ece80f85138
Found module libdbusmenu-glib.so.4 with build-id: c2fe215629a5a346cc5381408466d77f8c540e25
Found module libpcre.so.1 with build-id: 845483dd0acba86de9f0313102bebbaf3ce52767
Found module libresolv.so.2 with build-id: c915c72668282861a813f7ea3c0780f37b681dc0
Found module libmount.so.1 with build-id: d671e6fd6c9d6ec06604c1072754a431d02dc50c
Found module libz.so.1 with build-id: 81bf6e728a6d6f5b105b0f8b25f6c614ce10452a
Found module libXinerama.so.1 with build-id: dbb6adece34ea2766d3efcd628af8f9757344dd6
Found module libXrandr.so.2 with build-id: d4e869b5c72541e1de8f96b456248987ea52d51b
Found module libXcomposite.so.1 with build-id: f1bb5ffb6203ab2907ac044a204ce3902a79f1c1
Found module libXdamage.so.1 with build-id: a1d867e093c16944b2c0083be98050ef35c9e5c2
Found module libXcursor.so.1 with build-id: eb6315bcd7526aaf51b0a1307efdc587cea0a152
Found module libXext.so.6 with build-id: d70f24beb4fad748d6becffdcc13e51be0a2ebfa
Found module libwayland-egl.so.1 with build-id: 1c761f5ec5429abb88e4f6ff71815358e58c393c
Found module libwayland-cursor.so.0 with build-id: a8b45436314425ee9a92efaff68bd2b79f870e77
Found module libxkbcommon.so.0 with build-id: 9c297febc9583fd5b65e15f1bd89ec8172e97e50
Found module librt.so.1 with build-id: 75484da2d6f1515189eefa076e0a40328834cd16
Found module libXfixes.so.3 with build-id: 0a05c7e8714522bfbdd7c0027c3e2a94965664b0
Found module libcloudproviders.so.0 with build-id: ed1d9c9e60be4e858ad148484767249be1edf95f
Found module libatk-bridge-2.0.so.0 with build-id: 1161ebbcc5aa791075dfa026671875e5417287bd
Found module libX11.so.6 with build-id: e9192497aa39fab51358826d1a1b5fb77edc5081
Found module libXi.so.6 with build-id: 0557077390d87019cd9e0b3bb849643eaca73e30
Found module libepoxy.so.0 with build-id: 90da22e0a8d12c6b90fb00d95a23cc657b599334
Found module libatk-1.0.so.0 with build-id: aacaa40363d304858b654728a893bc212a4ca119
Found module libcairo-gobject.so.2 with build-id: abbf5a8a182ea1b8b7d68540df5c2e25c0bdae0c
Found module libfribidi.so.0 with build-id: e29e3b231f961d49e628d43ddd1237a8364f788c
Found module libfreetype.so.6 with build-id: 3131a701435f4d87afeab159b4aa57c4d151ffc3
Found module libfontconfig.so.1 with build-id: 1103a641395c7d3b42e49b793d3a9ea927c77bf6
Found module libpangoft2-1.0.so.0 with build-id: 704dc3cd81d2ea50cc4d8589b59c42ff6da55fa3
Found module libcairo.so.2 with build-id: 06820af9cf79c2deb207d9533a61caf04e67a8ad
Found module libharfbuzz.so.0 with build-id: 193a18a69d9e71d9988c6795c000fea51a2ac500
Found module libpango-1.0.so.0 with build-id: 68d673371ec32a28694d83dc9c8d6b348c91bb4e
Found module libpangocairo-1.0.so.0 with build-id: 2554be0474a6ee311b6e9aa4fd79a244167cdcfd
Found module libgmodule-2.0.so.0 with build-id: 1c305e1a08706be35604a78f8d99a567274a0217
Found module libgdk_pixbuf-2.0.so.0 with build-id: ca4beb15a30de8d8767f7bca8222038c61b58b36
Found module libatkmm-1.6.so.1 with build-id: de4113c984abd24cfd8bd4fd5bdb047cb1d3e1bc
Found module libpangomm-1.4.so.1 with build-id: 2cec36bcc7834108cb4d22aa33ee2082e8461eb3
Found module libcairomm-1.0.so.1 with build-id: d46392cba29606ecd937696fc4c4920d7943e197
Found module ld-linux-x86-64.so.2 with build-id: 040cc3dd10461562f177df39e3be2f3704258c3c
Found module libffi.so.7 with build-id: de60e99f39569d11d09160bbdcd486cedc87d2b6
Found module libc.so.6 with build-id: 4b406737057708c0e4c642345a703c47a61c73dc
Found module libpthread.so.0 with build-id: 07c8f95b4f3251d08550217ad8a1f31066229996
Found module libgcc_s.so.1 with build-id: 7f8508bb914546ada778809b64b99d234337d835
Found module libm.so.6 with build-id: 2b8fd1f869ecab4e0b55e92f2f151897f6818acf
Found module libstdc++.so.6 with build-id: 8ab0e57054dd1dcba681f217016afc6a4e639783
Found module libdate-tz.so.3 with build-id: 4c5d98c32642d954995a566297c1edb21cd1a2a7
Found module libsndio.so.7.1 with build-id: 902d3b3695b6b8d5550e2fef7604c814f254e462
Found module libgtk-layer-shell.so.0 with build-id: fe513eb7fe1d99817a894dcc1a6bd604157a23d8
Found module libmpdclient.so.2 with build-id: ff5214dacf7ecbd72851ad0770feee11facbd4ff
Found module libudev.so.1 with build-id: 880e57f50a36ddbfe80dbe5f62d43860f960c797
Found module libpulse.so.0 with build-id: a14b874b77e27c3f8349c1c219471b1fab12fe71
Found module libnl-genl-3.so.200 with build-id: f2475497c160d564dd619c097bad5b30dd786e31
Found module libnl-3.so.200 with build-id: 3e4695485d1726e09051f3d1abd0e2365a7c8a08
Found module libdbusmenu-gtk3.so.4 with build-id: 313794b4658942533b4d5cb971a6fb44f185ea54
Found module libglib-2.0.so.0 with build-id: 1683975c11bffb530743825149325e84765311ab
Found module libgobject-2.0.so.0 with build-id: a6d338dd0fd6bb335a4010b838f5173c6cbf953a
Found module libglibmm-2.4.so.1 with build-id: 245dd33982d8e182c8fa0adcbd81d235d80a1702
Found module libgio-2.0.so.0 with build-id: 451ca85a17a8e53e563a4a878f7f44901bd5a1e1
Found module libgdk-3.so.0 with build-id: 69fa63b702942fc002bbb4d367cd4423083523fd
Found module libgtk-3.so.0 with build-id: 7b16a75668848854c2613cd675e0cad10ed20a04
Found module libgiomm-2.4.so.1 with build-id: 2086eefc3fb3ebda0b85e2a34f4facf7a1c09d56
Found module libgdkmm-3.0.so.1 with build-id: 19b5880dd1579e371b4c8000089c3177311c5193
Found module libgtkmm-3.0.so.1 with build-id: e0f0b8d7c472550fde749fd1846242ef38e92104
Found module libjsoncpp.so.24 with build-id: 675605d3bda7f861069f5be2ee1f7004e5d90404
Found module libsigc-2.0.so.0 with build-id: 297c18f4158bceff37c96ba10b4c5b8b87bf22b1
Found module libspdlog.so.1 with build-id: a8d22e537a9cae8100f11987caafe6336002196e
Found module libfmt.so.8 with build-id: 1db611c72ff913efd050cf26b7dac304178e14a8
Found module libwayland-client.so.0 with build-id: 58038363d7ea1fd5e6532f6e5f90b1a3ce09388a
Found module waybar with build-id: c1a23ec8293d54cc6dadf4dd251767aa29015cd2
Stack trace of thread 86043:
#0  0x00007fa4e2c2cd22 raise (libc.so.6 + 0x3cd22)
#1  0x00007fa4e2c16862 abort (libc.so.6 + 0x26862)
#2  0x00007fa4e2fd7802 _ZN9__gnu_cxx27__verbose_terminate_handlerEv (libstdc++.so.6 + 0x99802)
#3  0x00007fa4e2fe3c8a _ZN10__cxxabiv111__terminateEPFvvE (libstdc++.so.6 + 0xa5c8a)
#4  0x00007fa4e2fe3cf7 _ZSt9terminatev (libstdc++.so.6 + 0xa5cf7)
#5  0x00007fa4e2fe35fd __gxx_personality_v0 (libstdc++.so.6 + 0xa55fd)
#6  0x00007fa4e2dee0b4 _Unwind_ForcedUnwind_Phase2 (libgcc_s.so.1 + 0x110b4)
#7  0x00007fa4e2dee7b2 _Unwind_ForcedUnwind (libgcc_s.so.1 + 0x117b2)
#8  0x00007fa4e2dce374 __pthread_unwind (libpthread.so.0 + 0x12374)
#9  0x00007fa4e2dc3c09 sigcancel_handler (libpthread.so.0 + 0x7c09)
#10 0x00007fa4e2dcf870 __restore_rt (libpthread.so.0 + 0x13870)
#11 0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#12 0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#13 0x000055d7c5598096 n/a (waybar + 0x67096)
#14 0x000055d7c556f561 n/a (waybar + 0x3e561)
#15 0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#16 0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#17 0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86029:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dc6743 __pthread_clockjoin_ex (libpthread.so.0 + 0xa743)
#2  0x00007fa4e3011628 __gthread_join (libstdc++.so.6 + 0xd3628)
#3  0x000055d7c559988e n/a (waybar + 0x6888e)
#4  0x000055d7c558058f n/a (waybar + 0x4f58f)
#5  0x000055d7c55807dc n/a (waybar + 0x4f7dc)
#6  0x000055d7c5585017 n/a (waybar + 0x54017)
#7  0x00007fa4e347c5d9 n/a (libglibmm-2.4.so.1 + 0x505d9)
#8  0x00007fa4e32ef015 g_main_context_dispatch (libglib-2.0.so.0 + 0x54015)
#9  0x00007fa4e3342b99 n/a (libglib-2.0.so.0 + 0xa7b99)
#10 0x00007fa4e32ec871 g_main_context_iteration (libglib-2.0.so.0 + 0x51871)
#11 0x00007fa4e357e8b6 g_application_run (libgio-2.0.so.0 + 0xcc8b6)
#12 0x000055d7c5558f8f n/a (waybar + 0x27f8f)
#13 0x000055d7c55582f6 main (waybar + 0x272f6)
#14 0x00007fa4e2c17b25 __libc_start_main (libc.so.6 + 0x27b25)
#15 0x000055d7c5559aae n/a (waybar + 0x28aae)

Stack trace of thread 86034:
#0  0x00007fa4e2ce3b2f __poll (libc.so.6 + 0xf3b2f)
#1  0x00007fa4e3342b29 n/a (libglib-2.0.so.0 + 0xa7b29)
#2  0x00007fa4e32ee693 g_main_loop_run (libglib-2.0.so.0 + 0x53693)
#3  0x00007fa4e35b38c8 n/a (libgio-2.0.so.0 + 0x1018c8)
#4  0x00007fa4e331d2c1 n/a (libglib-2.0.so.0 + 0x822c1)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86038:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c556d687 n/a (waybar + 0x3c687)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86037:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c559efc5 n/a (waybar + 0x6dfc5)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86042:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c55943f6 n/a (waybar + 0x633f6)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86036:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c559b5d5 n/a (waybar + 0x6a5d5)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86035:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c55a1f15 n/a (waybar + 0x70f15)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86051:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c55a1f15 n/a (waybar + 0x70f15)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86041:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c5574016 n/a (waybar + 0x43016)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86052:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c559b5d5 n/a (waybar + 0x6a5d5)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86030:
#0  0x00007fa4e2c2dae2 __sigtimedwait (libc.so.6 + 0x3dae2)
#1  0x00007fa4e2dcf3ac sigwait (libpthread.so.0 + 0x133ac)
#2  0x000055d7c557642c n/a (waybar + 0x4542c)
#3  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#4  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86053:
#0  0x00007fa4e2dcea2c recv (libpthread.so.0 + 0x12a2c)
#1  0x000055d7c559aa8c n/a (waybar + 0x69a8c)
#2  0x000055d7c559b08a n/a (waybar + 0x6a08a)
#3  0x000055d7c559efc5 n/a (waybar + 0x6dfc5)
#4  0x000055d7c556f561 n/a (waybar + 0x3e561)
#5  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#6  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#7  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86057:
#0  0x00007fa4e2ce3b2f __poll (libc.so.6 + 0xf3b2f)
#1  0x00007fa4e3237654 n/a (libpulse.so.0 + 0x33654)
#2  0x00007fa4e32209a9 pa_mainloop_poll (libpulse.so.0 + 0x1c9a9)
#3  0x00007fa4e322b281 pa_mainloop_iterate (libpulse.so.0 + 0x27281)
#4  0x00007fa4e322b331 pa_mainloop_run (libpulse.so.0 + 0x27331)
#5  0x00007fa4e323b7fe n/a (libpulse.so.0 + 0x377fe)
#6  0x00007fa4e22265cc n/a (libpulsecommon-14.2.so + 0x565cc)
#7  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#8  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86031:
#0  0x00007fa4e2ce3b2f __poll (libc.so.6 + 0xf3b2f)
#1  0x00007fa4e3342b29 n/a (libglib-2.0.so.0 + 0xa7b29)
#2  0x00007fa4e32ec871 g_main_context_iteration (libglib-2.0.so.0 + 0x51871)
#3  0x00007fa4e32ec8c2 n/a (libglib-2.0.so.0 + 0x518c2)
#4  0x00007fa4e331d2c1 n/a (libglib-2.0.so.0 + 0x822c1)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86058:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c5574016 n/a (waybar + 0x43016)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86054:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c556d687 n/a (waybar + 0x3c687)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86061:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb574 pthread_cond_timedwait@@GLIBC_2.3.2 (libpthread.so.0 + 0xf574)
#2  0x000055d7c55d45a6 n/a (waybar + 0xa35a6)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86040:
#0  0x00007fa4e2ce3b2f __poll (libc.so.6 + 0xf3b2f)
#1  0x00007fa4e3237654 n/a (libpulse.so.0 + 0x33654)
#2  0x00007fa4e32209a9 pa_mainloop_poll (libpulse.so.0 + 0x1c9a9)
#3  0x00007fa4e322b281 pa_mainloop_iterate (libpulse.so.0 + 0x27281)
#4  0x00007fa4e322b331 pa_mainloop_run (libpulse.so.0 + 0x27331)
#5  0x00007fa4e323b7fe n/a (libpulse.so.0 + 0x377fe)
#6  0x00007fa4e22265cc n/a (libpulsecommon-14.2.so + 0x565cc)
#7  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#8  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86033:
#0  0x00007fa4e2ce3b2f __poll (libc.so.6 + 0xf3b2f)
#1  0x00007fa4e3342b29 n/a (libglib-2.0.so.0 + 0xa7b29)
#2  0x00007fa4e32ec871 g_main_context_iteration (libglib-2.0.so.0 + 0x51871)
#3  0x00007fa4e0092ebe n/a (libdconfsettings.so + 0x5ebe)
#4  0x00007fa4e331d2c1 n/a (libglib-2.0.so.0 + 0x822c1)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86059:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c55943f6 n/a (waybar + 0x633f6)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)

Stack trace of thread 86060:
#0  0x00007fa4e2dd18ca __futex_abstimed_wait_common64 (libpthread.so.0 + 0x158ca)
#1  0x00007fa4e2dcb842 pthread_cond_clockwait (libpthread.so.0 + 0xf842)
#2  0x000055d7c5598096 n/a (waybar + 0x67096)
#3  0x000055d7c556f561 n/a (waybar + 0x3e561)
#4  0x00007fa4e30113c4 execute_native_thread_routine (libstdc++.so.6 + 0xd33c4)
#5  0x00007fa4e2dc5259 start_thread (libpthread.so.0 + 0x9259)
#6  0x00007fa4e2cee5e3 __clone (libc.so.6 + 0xfe5e3)
```

</details>

**Comment by stlaz** [comment]
I'm observing the same behavior on Arch

**Comment by Alexays** [comment]
Fixed on master/0.9.8
